### PR TITLE
Multiz and isolated ruins support.

### PR DESCRIPTION
**Pull request by AnturK**
Two new always_load_with options for ruins.
Use PLACE_ISOLATED to load the secondary ruin part in separate z level.
Use PLACE_BELOW to load it on the z level below.

### CODE REVIEW

File: code/controllers/subsystem/mapping.dm [hunk 1]
```diff
@@ -461,8 +463,9 @@ GLOBAL_LIST_EMPTY(the_station_areas)
 				return reserve
 		//If we didn't return at this point, theres a good chance we ran out of room on the exisiting reserved z levels, so lets try a new one
 		num_of_res_levels += 1
-		var/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
-		if(reserve.Reserve(width, height, newReserved))
+		var/datum/space_level/newReserved = add_new_zlevel("Transit/Reserved [num_of_res_levels]", list(ZTRAIT_RESERVED = TRUE))
+		initialize_reserved_level(newReserved.z_value)
```
Comment (AnturK) [inline code/controllers/subsystem/mapping.dm]: In case you're wondering yes this was broken before.

**Comment by Whoneedspacee** [comment]
does this mean that if i want it to always place below then i should spawn the top level ruins first so the bottom level ruins dont block it?

**Comment by AnturK** [comment]
Not sure if i understand the question, don't block what ?
Right now the validation of below z being appropriate is very basic, it's just checking if z actually exists.

**Comment by Whoneedspacee** [comment]
well if a ruin on a lower z level spawned in the same spot then it wouldn’t be able to spawn there would it?

right now for my alt lavaland project i have above ground ruins and below ground ruins

so if the below ground ruins spawned first they could possibly block the above grounds lower portion from spawning right?

**Comment by AnturK** [comment]
Good point, i will add toggling no ruins after successful load of ruin with lower part.

**Comment by Whoneedspacee** [comment]
yeah if below grounds toggled no ruin on the above part then above ruins with no below part wouldnt be able to spawn there either

better to spawn above grounds first anyways thanks for respondin

**Comment by Dennok** [comment]
Why not add PLACE_ABOVE too?

**Comment by AnturK** [comment]
There's really no difference what order you load them in and being only one directional makes handling conflicts easier.

**Pull request merged**